pkgs/development/python-modules/pylance/default.nix

```diff
@@ -0,0 +1,62 @@
+{ buildPythonPackage
+, Accelerate
+, CoreFoundation
+, Security
+, fetchFromGitHub
+, lib
+, libiconv
+, numpy
+, protobuf
+, pyarrow
+, rustPlatform
+, stdenv
+}:
+
+buildPythonPackage rec {
+  pname = "pylance";
+  version = "0.8.14";
+
+  src = fetchFromGitHub {
+    owner = "lancedb";
+    repo = "lance";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-egLpWxkSE0YOvJvvgmfcqq5SpCs8G6mXDNsLglBTmc8=";
+  };
+
+  sourceRoot = "${src.name}/python";
+
+  postPatch = ''
+    cp ${./Cargo.lock} Cargo.lock
+  '';
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    Accelerate
+    CoreFoundation
+    Security
+    libiconv
+    protobuf
+  ];
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  format = "pyproject";
```

Comment:
Should be `pyproject = true;` now (see https://github.com/NixOS/nixpkgs/commit/39c64e232328e09284c766d2b56f2d680e511d41).